service/storage_service.cc

```diff
@@ -3157,9 +3164,20 @@ future<> storage_service::replicate_to_all_cores(mutable_token_metadata_ptr tmpt
             }
 
             auto& table_erms = pending_table_erms[this_shard_id()];
+            auto& view_erms = pending_view_erms[this_shard_id()];
             for (auto it = table_erms.begin(); it != table_erms.end(); ) {
                 auto& cf = db.find_column_family(it->first);
                 cf.update_effective_replication_map(std::move(it->second));
+                for (const auto& view_ptr : cf.views()) {
+                    const auto& view_id = view_ptr->id();
+                    auto view_it = view_erms.find(view_id); 
+                    if (view_it == view_erms.end()) {
+                        throw std::runtime_error(format("Could not find pending effective_replication_map for view {}.{} id={}", view_ptr->ks_name(), view_ptr->cf_name(), view_id));
+                    }
+                    auto& view = db.find_column_family(view_id);
+                    view.update_effective_replication_map(std::move(view_it->second));
+                    view_erms.erase(view_it);
+                }
```

Comment:
You're right.
Now that we prevent this yield on purpose I can remove that test scenario and instead add a deliberate maybe_yield call here to prevent stalls with many tables.

Reply:
Done in v3 ([2f1bb11](https://github.com/scylladb/scylladb/commit/2f1bb11548d8796c227cf0ad2e3e2304631ac6ba))